rust/blockstore/src/arrow/blockfile.rs

```diff
@@ -102,6 +102,10 @@ impl ArrowBlockfileWriter {
                 removed = self.root.sparse_index.remove_block(&delta.id);
             }
             if !removed {
+                self.root
```

Comment:
Since this is only set for the blocks that this run of compaction touches, what happens when compaction is run for the first time after this change is merged (i.e. with new code)? Will compaction only set counts for some of the blocks and thus the count arrow array will not have the same size as the block uuid array causing some sort of panic?

Reply:
Yes I had this same realization last night, I will fix and add a test

Reply:
https://github.com/chroma-core/chroma/pull/2938/commits/f0612ec312889d7714a01e71d15aaff043e9f048